common/ferrostar/src/models.rs

```diff
@@ -12,17 +12,16 @@ use alloc::{string::String, vec::Vec};
 use geo::{Coord, LineString, Point, Rect};
 #[cfg(feature = "uniffi")]
 use polyline::encode_coordinates;
-use serde::Deserialize;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
 
 #[cfg(all(feature = "std", not(feature = "web-time")))]
 use std::time::SystemTime;
 
 #[cfg(feature = "web-time")]
 use web_time::SystemTime;
 
-#[cfg(any(test, feature = "wasm-bindgen"))]
-use serde::Serialize;
-
+use std::collections::BTreeMap;
```

Comment:
This is primarily driven by snapshot testing. Using a `HashMap` we'd have to rethink how we test all of the route deserialization stuff. It also helps visually compare things when you're looking a route step because the data will maintain it's column (not required but a nice to have).

Reply:
Hmmm.... I feel like data structure selection based on how they look in snapshot tests is a slippery slope. Let's see if we we can find a way to make the test output comparable. I'm a bit surprised that insta doesn't already account for this already TBH as `HashMap` is the go-to data structure for this sort of thing.

Reply:
The issue is we're asserting after converting back to a string (or bytes if desired). So maybe we can restructure the process a bit to snapshot while it's still a hashmap.

Reply:
Ok; let's add that to the discussion agenda for our next meeting. Probably easier to talk through it live and maybe I'll try live coding a few ideas.

Reply:
Reverted to `HashMap` and added:

1. `sort_maps` insta setting for tests that validate while it's a HashMap. See https://docs.rs/insta/1.40.0/insta/struct.Settings.html
2. Redact the annotations elsewhere.